src/java/test/org/apache/zookeeper/client/ZKClientConfigTest.java

```diff
@@ -0,0 +1,141 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.apache.zookeeper.server.quorum.QuorumPeerConfig.ConfigException;
+import org.junit.Test;
+
+public class ZKClientConfigTest {
+
+    @Test
+    public void testDefaultConfiguration() {
+
+        String[] values = new String[] { "zookeeper1", "Client1", "true", "zookeeper/hadoop.hadoop.com", "true",
+                "ClientCnxnSocketNetty", "true" };
+        String[] properties = new String[] { ZK_SASL_CLIENT_USERNAME, LOGIN_CONTEXT_NAME_KEY, ENABLE_CLIENT_SASL_KEY,
+                ZOOKEEPER_SERVER_REALM, DISABLE_AUTO_WATCH_RESET, ZOOKEEPER_CLIENT_CNXN_SOCKET, SECURE_CLIENT };
+        assertEquals(properties.length,values.length);
+        for (int i = 0; i < properties.length; i++) {
+            String prop = properties[i];
+            System.setProperty(prop, values[i]);
+        }
+        /**
+         * ZKClientConfig should get initialized with system properties
+         */
+        ZKClientConfig conf = new ZKClientConfig();
+        for (int i = 0; i < properties.length; i++) {
+            String prop = properties[i];
+            String result = conf.getProperty(prop);
+            assertEquals(values[i], result);
+        }
+
+        /**
+         * clear properties
+         */
+        for (String prop : properties) {
+            System.clearProperty(prop);
+        }
+
+        conf = new ZKClientConfig();
+        /**
+         * test that all the properties are null
+         */
+        for (String prop : properties) {
+            String result = conf.getProperty(prop);
+            assertNull(result);
+        }
+
+    }
+
+    @Test
+    public void testSystemProprtyValue() {
+        String clientName = "zookeeper1";
+        System.setProperty(ZK_SASL_CLIENT_USERNAME, clientName);
+
+        ZKClientConfig conf = new ZKClientConfig();
+        assertEquals(conf.getProperty(ZK_SASL_CLIENT_USERNAME), clientName);
+
+        String newClientName = "zookeeper2";
+        conf.setProperty(ZK_SASL_CLIENT_USERNAME, newClientName);
+
+        assertEquals(conf.getProperty(ZK_SASL_CLIENT_USERNAME), newClientName);
+    }
+
+    @Test
+    public void testReadConfigurationFile() throws IOException, ConfigException {
+        File file = new File("test.conf");
+        OutputStream io = new FileOutputStream(file);
+        try {
+            io.write((ENABLE_CLIENT_SASL_KEY + "=true\n").getBytes());
+            io.write((ZK_SASL_CLIENT_USERNAME + "=ZK\n").getBytes());
+            io.write((LOGIN_CONTEXT_NAME_KEY + "=MyClient\n").getBytes());
+            io.write((ZOOKEEPER_SERVER_REALM + "=HADOOP.COM\n").getBytes());
+            io.write(("dummyProperty=dummyValue").getBytes());
+            io.flush();
```

Comment:
It doesn't hurt to have a call to ``flush`` here (I __guess__), but the closing of ``FileOutputStream`` at line 109 will flush the pending content to disk before eventually close the file descriptors.